packages/types/index.js

```diff
@@ -0,0 +1,7 @@
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: @loopback/types
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+const nodeMajorVersion = +process.versions.node.split('.')[0];
+module.exports = nodeMajorVersion >= 7 ? require('./dist') : require('./dist6');
```

Comment:
We no longer support Node.js 6.x, where does this line come from? Could you please use the latest project infrastructure, e.g. as scaffolded by `lb4 extension`?

Reply:
This PR builds on top of #947, which was proposed before we dropped Node 6. I didn't bother with pretty easily fixable stuff like this one to focus on the subject of the PoC itself. If we decide to go forward with the PoC, I (or someone else building on top of the PoC) will definitely need to fix this.

Reply:
Agreed 👍